R/html_dependency.R

```diff
@@ -505,6 +525,50 @@ renderDependencies <- function(dependencies,
   HTML(paste(html, collapse = "\n"))
 }
 
+
+
+renderScript <- function(script, srcpath, encodeFunc, hrefFilter) {
+  # If the input is a named list, transform it to an unnamed list
+  # whose only element is the input list
+  if (anyNamed(script)) {
+    if (anyUnnamed(script)) stop("script inputs cannot mix named and unnamed")
+    script <- list(script)
+  }
+
+  # For each element, if it's a scalar string, transform it to a named
+  # list with one element, "src".
+  script <- lapply(script, function(item) {
+    if (length(item) == 1 && is.character(item)) {
+      item = list(src = item)
+    }
+
+    if (length(names(item)) == 0) {
+      stop(
+        "Elements of script must be named lists, or scalar strings ",
+        "I got ", deparse(item)
+      )
+    }
+
+    return(item)
+  })
+
+  script <- vapply(
+    script, function(x) {
+      x$src <- hrefFilter(file.path(srcpath, encodeFunc(x$src)))
+      paste0(
+        "<script", 
+        paste0(" ", names(x), '="', x, '"', collapse = ''),
```

Comment:
Sorry, I forgot to mention that there should be `htmlEscape(x)` here.

Reply:
It would be good if one of the tests had a character that needs escaping, to make sure this doesn't get overlooked if there are changes in the future.

Reply:
I think those test cases are a very good idea, it's easy behavior to regress and dangerous if allowed to happen, 

however i think they constitute a good chunk of testing, and kind of comprise their own unit of testing, so i wont be adding that to this PR

Reply:
Changes implemented

Reply:
FWIW, I think just that if you modify one test case with a weird character, that's probably sufficient for our purposes.

Reply:
Resolved, added a brand new test case with an explicit purpose though